Arm backend: Make upstream testing more target specific

Michiel-Olieslagers opened the pull request:
This patch splits up our upstream testing such that only relevant items are installed and only required things are tested upstream. It removes any duplicate testing and makes the upstream testing structure more transparent and readable.

Change-Id: I14cf329eed03d9859aef97204a9f12185b9533a2


cc @freddan80 @per @zingo @oscarandersson8218 @digantdesai

Michiel-Olieslagers committed:
Make upstream testing more target specific

This patch splits up our upstream testing such that only relevant items are installed and only
required things are tested upstream. It removes any duplicate testing and makes the upstream
testing structure more transparent and readable.

(Rebased & fixed certain syntactical issues in CI tests)

Change-Id: I14cf329eed03d9859aef97204a9f12185b9533a2

Michiel-Olieslagers committed:
Fix and update run.sh script for VGF target.

Change-Id: I695c2563b3f1ebcbd6b6eb660dff36c305c0325a

metascroy commented:
> @metascroy CI seen to be fixed now and I rebased to retrigget job, now Meta internal and rebased version differ and we need help to re-import, hope that is OK

Re-imported

zingo commented:
Oh no :( @Michiel-Olieslagers  another bad named test case has sneeked passed this PR

```
def test_fail_dump_tosa_ops(capsys, test_data: input_t1):
   ...
    pipeline = EthosU55PipelineINT[input_t1](...
```
    
  I do not know why the test-arm-ootb-linux tests fails :(

Michiel-Olieslagers committed:
Arm backend: Fixed scratch-dir naming issues

Fixed call to wrong location for arm-scratch-dir and changed
no_target function behaviour.

Change-Id: Idf292ee747ffcf8ffed13d725849cc686d4c60af
Signed-off-by: Michiel Olieslagers <michiel.olieslagers@arm.com>

Michiel-Olieslagers committed:
Merge branch 'main' into change-1126014

zingo committed:
Merge branch 'main' into change-1126014

zingo committed:
Merge branch 'main' into change-1126014

Michiel-Olieslagers committed:
Merge branch 'main' into change-1126014

digantdesai reviewed (approved):
Review automatically exported from Phabricator review in Meta.
